forge-ai/src/main/java/forge/ai/ComputerUtil.java

```diff
@@ -2572,7 +2572,13 @@ else if (logic.equals("MostProminentInComputerGraveyard")) {
                 }
             }
             if (!CardType.isACreatureType(chosen) || invalidTypes.contains(chosen)) {
-                chosen = "Sliver";
+                // assume invalid types is all non-valid types
+                List <String> invalid = Lists.newArrayList(CardType.getAllCreatureTypes());
```

Comment:
I don't understand the question 100%?
tag it in the SpellAbility AILogic?

Reply:
well moving away from the hardcoded default is probably a good idea
but if you just move the lines after `logic != null` branch out of it you can reuse them here so only L2580 should be needed

Reply:
So the whole `if` block?
I used `getMostProminentType` as a decision making tool, since you might as well note the types with the most first.

Making a script tag wouldn't be bad since only 2 cards currently use this singleton noted creature types abilitiy

Reply:
I completely reduced the code there. `invalid` wasn't used. Also added check for `validType.size() == 1` skipping get most prominent type